### ISSUE TITLE
Chapitre sur les tableaux - Pistes d'améliorations

### ISSUE THREAD

**Issue by pgonzalezalv** [OpenWeek/java-syllabus-pages#5]
Dans la partie théorie, il manque :

* plus de détails sur comment déterminer la taille d'un tableau 1D, 2D ou à plus de deux dimensions (`length`);
* la différence entre `length` et la méthode `length()`.

De manière générale, il faudrait peut-être séparer la théorie en plusieurs pages mais avec des qcm et/ou des exercices simples juste à la fin de chaque page de théorie.

**Comment by AlexandreDubray** [comment]
Hello,

pour la différence entre ``length`` et ``length()``,  je suppose que tu fais référence à la différence entre attribut et méthode de classe? Si oui, alors en parler sans avoir vu les objets semble un peu compliqué non?

Pour la division du chapitre, À quelle répartition penses-tu exactement? Quelques chose du genre

1. Tableau à une dimension: initialisation
2. Tableau à une dimension: Opération fréquente (itération dessus, méthode de taille, etc)
3. Tableau à plusieurs dimensions

Avec chaque fois des sous-section bien spécifique? Par exemple, déterminer la taille d'une dimension d'un tableau multi-dimensionnel? On pourrait alors imaginer un exercice qui consiste à afficher un tableau à deux dimensions. Simple mais pratique pour se représenter la structure d'un tableau à deux dimension.

**Comment by pgonzalezalv** [comment]
Oui, tu as raison : c'est compliqué de parler de `length()` à ce stade!

Pour la répartition j'avais la même idée, mais les titres suivants : 

1. Initialiser un tableau
2. Opérations sur un tableau
3. Tableaux multidimensionnels

Pour les détails (qcm, exercices), je vais travailler ça petit à petit cette semaine et je PR après.

A oui, je PR sur le repo d'[ALambot](https://github.com/ALambot/java-syllabus-pages) du coup, ou tu vas accepter son PR (et du coup je ferai directement ici le PR)?

**Comment by Jimvy** [comment]
A propos : QCM question 10, toutes les réponses sont justes parce qu'il manque un .. negative-multiple quelque part...

**Comment by Jimvy** [comment]
Exercices INGInious, question 5 : comme la tâche INGInious requiert deux exercices, il y a une boucle infinie qui survient car on ne fournit qu'un seul exercice.

**Comment by AlexandreDubray** [comment]
@pgonzalezalv Il y a déjà une partie de la théorie sur les tableaux qui se trouve dans la PR de d'@ALambot non? Si oui c'est peut-être plus cohérent de continuer de travailler la dessus non? Sinon tu peux faire une PR à part, elles seront toutes regardée et mergée dans les semaines qui arrivent :-) Pour les titres, ok pour moi

**Comment by pgonzalezalv** [comment]
@Jimvy oui, j'avais vu pour le QCM, mais je n'ai pas encore PR sur le repo d'[ALambot](https://github.com/ALambot/java-syllabus-pages). Merci!

Je vais revoir ça pour les exercices, merci aussi (j'avais repris, sans trop réfléchir, ce qu'on avait fait l'année passée à l'OW2016).

@AlexandreDubray Oui, tout à fait! En fait, j'ai 2 remotes en plus en local (le repo de ALambot et celui-ci) comme ça quand je PR, vous avez rien à modifier (vu que je fetch avant de PR). Je posais la question pour éviter qu'il doive faire un PR vers celui-ci, si je PR sur le sien :)

**Comment by francoismichel** [comment]
@pgonzalezalv Il faudrait aussi transformer l'exercice `Modifie le code suivant pour créer un tableau vide de taille 10 :` de la page de théorie en un exercice INGInious :-)

Dans la section **Tableaux et boucles**, il manque un commentaire pour le `(3)`

Sinon bien joué Pablo ! :-)
